tracing-subscriber/src/filter/targets.rs

```diff
@@ -344,20 +379,111 @@ impl<S> layer::Filter<S> for Targets {
     }
 }
 
+impl IntoIterator for Targets {
+    type Item = (String, LevelFilter);
+
+    type IntoIter = IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        fn unpack(directive: StaticDirective) -> Option<(String, LevelFilter)> {
+            let level = directive.level;
+            directive.target.map(move |target| (target, level))
+        }
```

Comment:
just a note: this shouldn't be _strictly_ necessary --- the closure would have still worked, as a closure that doesn't borrow anything can be cast to a `fn`. but, this might be more readable than a cast! so, looks good to me as is.

Reply:
TIL! That indeed works. Decided to switch to closures to avoid repeating the signature.